Revert "drawio: update license"

Luflosi opened the pull request:
This reverts commit da2060c0dee86a677096aab104c029a578bd12ea.
See https://github.com/jgraph/drawio/commit/5b2e73471e4fea83d681f0cec5d1aaf7c3884996.

Cc @alyssais.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Luflosi committed:
drawio: update license

The LICENSE file of https://github.com/jgraph/drawio claims Apache License Version 2.0 again since https://github.com/jgraph/drawio/commit/5b2e73471e4fea83d681f0cec5d1aaf7c3884996.
But the README says:
> The minified code authored by us in this repo is licensed under an Apache v2 license, but the sources to build those files are not in this repo. This is not an open source project.

crabdancing commented:
Is this a good idea?

The readme says:

```
The minified code authored by us in this repo is licensed under an Apache v2 license, but the sources to build those files are not in this repo. This is not an open source project.
```

It seems like quite a hack if someone can just, e.g., compile a binary and then declare that binary to be GNU GPLv3, but you have to use a disassembler and a debugger. I would consider that a special case of a general principle that, for something to qualify for an 'open source' license, source code that's directly _intended for editing_ ought to be available.

With scripted languages that can be minified vs whitespaced, commented, and clearly labeled -- the line between source code and build product can get a little harier, but I think NixOS ought to err on the side of this sort of thing being considered `unfree`. What do you think?

dvaerum commented:
> but I think NixOS ought to err on the side of this sort of thing being considered `unfree`. What do you think?

I agree, I also think this should stay/remain marked as unfree. You don't have the Source Code if only the "minified code" is release, which I would consider unfree no matter the license.

Luflosi commented:
Thank you for noticing.
I also agree.

I changed `licenses.unfree` to `licenses.unfreeRedistributable` since I feel it better describes the situation while still requiring explicit opt-in from users.
I added `asl20` as a second license since it is technically still involved. This does not change the opt-in behavior. If you feel that I should remove it again, please tell me.
I also added a comment in the code explaining the situation so nobody naive like me tries to change it to a free license again.
